src/UseEffect/AntiPatterns/Examples/ApiRequest.tsx

```diff
@@ -7,14 +6,21 @@ interface Props {
 
 export function ApiRequestAntiPattern(props: Props) {
   const { firstName } = props;
-  const [localFirstName, setLocalFirstName] = useState(firstName);
-  const { data, isLoading } = useQuery<string>(firstName);
+  const {
+    data: localFirstName,
+    isLoading,
+    error,
+  } = useQuery<string>(firstName, {
+    onError: () => {
+      // show error toast
+    },
+    // onSuccess...
```

Comment:
This is an example where we should do things imperatively.
In other words - When an API call fails, imperatively show a toast.
(do not add a `useEffect` dependent on `error`)